src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -575,44 +575,25 @@ impl<'a, 'tcx> MatchCheckCtxt<'a, 'tcx> {
     }
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq)]
 enum Constructor<'tcx> {
     /// The constructor of all patterns that don't vary by constructor,
     /// e.g., struct patterns and fixed-length arrays.
     Single,
     /// Enum variants.
     Variant(DefId),
     /// Literal values.
-    ConstantValue(&'tcx ty::Const<'tcx>, Span),
-    /// Ranges of literal values (`2..=5` and `2..5`).
-    ConstantRange(u128, u128, Ty<'tcx>, RangeEnd, Span),
+    ConstantValue(&'tcx ty::Const<'tcx>),
+    /// Ranges of integer literal values (`2`, `2..=5` or `2..5`).
+    IntRange(IntRange<'tcx>),
+    /// Ranges of non-integer literal values (`2.0..=5.2`).
```

Comment:
This is specifically floating-point ranges now, as far as I'm aware, so maybe we should call this `FloatRange` instead.